### ISSUE TITLE
Allow override of messages in new admin/account consoles without replacing the entire .json file

### ISSUE THREAD

**Issue by xgp** [keycloak/keycloak#22791]
### Description

Allow overriding only some keys in the i18n messgaes .json files when creating a theme for the new admin/account consoles.

### Motivation

The changes to the mechanism of i18n in the new admin/account consoles mean that overriding a subset of the keys is no longer possible. Per https://github.com/keycloak/keycloak/issues/22431#issuecomment-1697083332 when creating a new theme, it is required to copy the whole .json file and replace the values you need. This means that theme authors will need to update these .json files for each release in case changes were made and new keys were added.

### Details

Update the mechanism of serving the .json messages files to merge files rather than replace them. This would make it possible to only replace a single key in a new theme. E.g. the file `theme/mycompany/admin/resources/locales/en/users.json` with the content
```json 
{
    "deleteConfirmCurrentUser": "Are you sure you want to permanently delete this user foobar"
}
```
would only replace the `deleteConfirmCurrentUser` key, and the defaults would be used for all other keys.

cc @jonkoops

**Comment by jonkoops** [comment]
Yeah, I agree this should be something that needs to be possible. The question is how we'd like to implement this. Perhaps we'd like to allow the user to specify labels that should be overwritten in a properties file, and then inject these into the template so they can be extracted.

**Comment by xgp** [comment]
@jonkoops :+1: 

When I was thrashing around trying to figure out if it was already implemented, I tried `users.deleteConfirmCurrentUser=...` in the messages properties file, which seemed like a reasonable way to do it.

**Comment by ssilvert** [comment]
@edewit @jonkoops I think this works already for properties files.  We just need to use the same logic in our json files.

As we create the "effective message bundle", it just needs to take into account that there is a hierarchy of themes with the child theme always taking precedence over the parent.

**Comment by jonkoops** [comment]
@ssilvert Yeah agreed. However I think we should align this so the translations are all contained in property files as folks have been used to.

**Comment by jonkoops** [comment]
I believe that this might be partially fixed by #24316, at least for the Admin Console.

**Comment by ssilvert** [comment]
> I believe that this might be partially fixed by #24316, at least for the Admin Console.

I think you are probably right.

@edewit Can you confirm?  If you think it's now fixed maybe @xgp can check it out in the nightly build.

**Comment by edewit** [comment]
Closing as this is now fixed
